### ISSUE TITLE
Provide esm version

### ISSUE THREAD

**Issue by tobiasdiez** [apollographql/apollo-utils#257]
With more and more build tools (vite, nuxt, etc) going towards an esm-first build, it would be nice if a proper [esm interface](https://nodejs.org/api/esm.html) could be provided, i.e. transpile and distribute esm version, including correct [export](https://nodejs.org/api/packages.html#conditional-exports) declaration in `package.json`.

**Comment by trevor-scheer** [comment]
@tobiasdiez yeah, I think we knew this should happen eventually - thanks for the nudge. Got a PR open for it. I'll test out the builds next week, should be able to get it merged and published in a number of days. If you don't mind trying out the builds that would be great!

**Comment by floydspace** [comment]
Hey @tobiasdiez I'd like to nudge again. this fix would cut 100kb from my app bundle size. Thank you

**Comment by trevor-scheer** [comment]
@floydspace the changes on the linked PR take after Apollo Server's approach, which has been updated once already and is [awaiting another fix](https://github.com/apollographql/apollo-server/issues/7625) from a contributor who has offered to implement the change. I'd like to wait until that change lands, then I can follow up with an equivalent update on #259.

**Comment by d3c0d3dpt** [comment]
Hey @trevor-scheer, I see that the other fix is nnot really moving for quite a while. Any chance we can get the ball running on this one?

**Comment by trevor-scheer** [comment]
My #259 PR follows Apollo Server's pattern. I don't think Apollo Server has it right currently, so I'm reluctant to adopt the same pattern until we figure out the right one. I unfortunately haven't had time to investigate this and I'm not sure when that might happen.
Ref: https://github.com/apollographql/apollo-server/issues/7625

If anyone feels motivated to solve the TS + CJS/ESM dual-package problem in a monorepo I would love to see that. The results of my limited investigation up to this point are documented in the linked issue. There's potentially some tooling that could help us there if we bend it to our will but I haven't tried anything out yet.
